### PR TITLE
library clean-up: Don't build libNX_Xdamage anymore. Use system's libXdamage shared library. (Fixes ArcticaProject/nx-libs#6, X2GoBTS#826).

### PR DESCRIPTION
**Pull request by sunweaver**
The (NX_)Xdamage library gets used in libXcompshad3 and provides the visual part of the shadowing support in NX.

It turned out that nxagent itself is not using any of the (NX_)Xdamage symbols, so linking nxagent directly against Xdamage has been dropped in this commit.

### CODE REVIEW

File: nx-X11/include/extensions/Imakefile [hunk 1]
```diff
@@ -47,9 +47,6 @@ RENDERHEADERS = render.h renderproto.h
 #if BuildXfixes || BuildXfixesLibrary
 XFIXESHEADERS = xfixeswire.h xfixesproto.h
 #endif
-#if BuildDamage || BuildDamageLibrary
-DAMAGEHEADERS = damagewire.h damageproto.h
```
Comment (Ionic) [inline nx-X11/include/extensions/Imakefile:51]: Both files can probably be removed from `debian/copyright.in` (although I remember we've been talking about that... but not the outcome of this conversation. Leaving it in copyright.in is probably fine anyway.)

File: nx-X11/programs/Xserver/Imakefile [hunk 2]
```diff
@@ -427,7 +427,7 @@ $(NXAGENTOBJS) $(NXAGENTLIBS) $(NXAGENTSYSLIBS):: $(NXAGENTDIRS)
 #if defined(SunArchitecture)
 NXAGENTNXLIBS = -L ../../../nxcomp -L ../../../nxcompext -L ../../../nxcompshad \
                 -lXcomp -lXcompext -lXcompshad -lrt -L/usr/sfw/lib -lNX_Xrender -lNX_Xfixes -lXfont \
-                -L../../../nx-X11/exports/lib -lNX_Xtst -lNX_Xdamage -lNX_Xrandr -lNX_Xcomposite -lXdmcp \
```
Comment (Ionic) [inline nx-X11/programs/Xserver/Imakefile:430]: Removing... or replacing with `-lXdamage`?

File: nx-X11/programs/Xserver/Imakefile [hunk 3]
```diff
@@ -437,7 +437,7 @@ NXAGENTNXLIBS = -L ../../../nxcomp -L ../../../nxcompext \
 #else
 NXAGENTNXLIBS = -L ../../../nxcomp -L ../../../nxcompext -L ../../../nxcompshad \
                 -lXcomp -lXcompext -lXcompshad -lNX_Xrender -lNX_X11 -lNX_Xext -lNX_Xfixes -lXfont \
-                -L../../../nx-X11/exports/lib -lNX_Xtst -lNX_Xdamage -lNX_Xrandr -lNX_Xcomposite -lNX_Xinerama -lXdmcp \
```
Comment (Ionic) [inline nx-X11/programs/Xserver/Imakefile:440]: Same here.

**Comment by Ionic** [comment]
Looks like I don't have many nits this time. Hopefully I didn't overlook something.

File: nx-X11/include/extensions/Imakefile (hunk 1 above)
Comment (sunweaver) [inline nx-X11/include/extensions/Imakefile:51]: On  Di 12 Mai 2015 05:45:34 CEST, Mihai Moldovan wrote:

> > @@ -47,9 +47,6 @@ RENDERHEADERS = render.h renderproto.h
> >  #if BuildXfixes || BuildXfixesLibrary
> >  XFIXESHEADERS = xfixeswire.h xfixesproto.h
> >  #endif
> > -#if BuildDamage || BuildDamageLibrary
> > -DAMAGEHEADERS = damagewire.h damageproto.h
> 
> Both files can probably be removed from `debian/copyright.in`  
> (although I remember we've been talking about that... but not the  
> outcome of this conversation. Leaving it in copyright.in is probably  
> fine anyway.)

debian/copyright.in is script-created. We will re-run this script  

## prior to a release.

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
fon: +49 (1520) 1976 148

GnuPG Key ID 0x25771B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

freeBusy:
https://mail.das-netzwerkteam.de/freebusy/m.gabriel%40das-netzwerkteam.de.xfb

File: nx-X11/programs/Xserver/Imakefile (hunk 2 above)
Comment (sunweaver) [inline nx-X11/programs/Xserver/Imakefile:430]: On  Di 12 Mai 2015 05:49:40 CEST, Mihai Moldovan wrote:

> > @@ -427,7 +427,7 @@ $(NXAGENTOBJS) $(NXAGENTLIBS)  
> > $(NXAGENTSYSLIBS):: $(NXAGENTDIRS)
> >  #if defined(SunArchitecture)
> >  NXAGENTNXLIBS = -L ../../../nxcomp -L ../../../nxcompext -L  
> > ../../../nxcompshad \
> >                  -lXcomp -lXcompext -lXcompshad -lrt -L/usr/sfw/lib  
> > -lNX_Xrender -lNX_Xfixes -lXfont \
> > -                -L../../../nx-X11/exports/lib -lNX_Xtst  
> >   -lNX_Xdamage -lNX_Xrandr -lNX_Xcomposite -lXdmcp \
> 
> Removing... or replacing with `-lXdamage`?

Removing. nxagent has been overlinked with NX_Xdamage, so far. Xdamage  

## linking is only needed in libXcompshad.so.3.

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
fon: +49 (1520) 1976 148

GnuPG Key ID 0x25771B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

freeBusy:
https://mail.das-netzwerkteam.de/freebusy/m.gabriel%40das-netzwerkteam.de.xfb

**Comment by sunweaver** [comment]
On  Di 12 Mai 2015 05:52:17 CEST, Mihai Moldovan wrote:

> Looks like I don't have many nits this time. Hopefully I didn't  
> overlook something.

## :-) I replied to the other comments. Please feel free to merge then...

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
fon: +49 (1520) 1976 148

GnuPG Key ID 0x25771B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

freeBusy:
https://mail.das-netzwerkteam.de/freebusy/m.gabriel%40das-netzwerkteam.de.xfb

**Comment by Ionic** [comment]
But that's actually something I should merge back into 3.5.0.x, right?

**Comment by sunweaver** [comment]
On  Di 12 Mai 2015 06:09:49 CEST, Mihai Moldovan wrote:

> But that's actually something I should merge back into 3.5.0.x, right?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ArcticaProject/nx-libs/pull/18#issuecomment-101124015

IMHO: No. I would not start merging back the code removal parts from  
this branch. I'd even more say, let's focus on backporting fixes for  
CVE issues and big problems.

I am working on a build env for Arctica, so we will have binary builds  
out soon. X2Go users could then pull-in nx-libs from another location  
(or you mirror the nx-libs build for Arctica for a while).

I am also thinking of releasing the first nx-libs 3.5.99.0 version  
sooner than orginally thought and approach the Debian release and  
security teams with that reduced code tree. It may be time for X2Go to  
change over then, as well. (Sometime around July, probably).

## 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
fon: +49 (1520) 1976 148

GnuPG Key ID 0x25771B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

freeBusy:
https://mail.das-netzwerkteam.de/freebusy/m.gabriel%40das-netzwerkteam.de.xfb

**Comment by Ionic** [comment]
But doesn't it fix Bug#826? Should this bug just stay open until X2Go can switch to 3.6.x?

**Comment by sunweaver** [comment]
> But doesn't it fix Bug#826? Should this bug just stay open until X2Go
> can switch to 3.6.x?

That's what I think. Mark it as 

Control: fixed -1 3.5.99.0
Control: tags -1 pending

on X2Go BTS and that's it. The other option would be disecting lXdamage (X.Org Git) and see what of their commits may be a fix for #826. This commit could then be cherry-picked over to nx-X11/lib/Xdamage. But, personally, I'd not bother as the use case is rare. A fix is in sight...

**Pull request merged**